libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/ui/SalesforceDroidGapActivity.kt

```diff
@@ -0,0 +1,604 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.phonegap.ui
+
+import android.net.Uri.parse
+import android.os.Bundle
+import android.view.KeyEvent
+import android.webkit.URLUtil.isHttpsUrl
+import androidx.lifecycle.lifecycleScope
+import com.salesforce.androidsdk.app.SalesforceSDKManager
+import com.salesforce.androidsdk.auth.HttpAccess.NoNetworkException
+import com.salesforce.androidsdk.config.BootConfig
+import com.salesforce.androidsdk.config.BootConfig.getBootConfig
+import com.salesforce.androidsdk.config.BootConfig.isAbsoluteUrl
+import com.salesforce.androidsdk.config.BootConfig.validateBootConfig
+import com.salesforce.androidsdk.config.LoginServerManager.PRODUCTION_LOGIN_URL
+import com.salesforce.androidsdk.config.LoginServerManager.SANDBOX_LOGIN_URL
+import com.salesforce.androidsdk.phonegap.app.SalesforceHybridSDKManager
+import com.salesforce.androidsdk.phonegap.ui.SalesforceWebViewClientHelper.getAppHomeUrl
+import com.salesforce.androidsdk.phonegap.ui.SalesforceWebViewClientHelper.hasCachedAppHome
+import com.salesforce.androidsdk.phonegap.util.SalesforceHybridLogger.i
+import com.salesforce.androidsdk.phonegap.util.SalesforceHybridLogger.w
+import com.salesforce.androidsdk.rest.ApiVersionStrings.VERSION_NUMBER
+import com.salesforce.androidsdk.rest.ClientManager
+import com.salesforce.androidsdk.rest.RestClient
+import com.salesforce.androidsdk.rest.RestClient.AsyncRequestCallback
+import com.salesforce.androidsdk.rest.RestRequest
+import com.salesforce.androidsdk.rest.RestRequest.getCheapRequest
+import com.salesforce.androidsdk.rest.RestResponse
+import com.salesforce.androidsdk.ui.SalesforceActivityDelegate
+import com.salesforce.androidsdk.ui.SalesforceActivityInterface
+import com.salesforce.androidsdk.util.AuthConfigUtil.MyDomainAuthConfig
+import com.salesforce.androidsdk.util.AuthConfigUtil.getMyDomainAuthConfig
+import com.salesforce.androidsdk.util.EventsObservable
+import com.salesforce.androidsdk.util.EventsObservable.EventType.GapWebViewCreateComplete
+import kotlinx.coroutines.launch
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import org.apache.cordova.CallbackContext
+import org.apache.cordova.CordovaActivity
+import org.apache.cordova.CordovaWebView
+import org.apache.cordova.CordovaWebViewEngine
+import org.apache.cordova.CordovaWebViewImpl.createEngine
+
+/**
+ * Class that defines the main activity for a PhoneGap-based application.
+ */
+open class SalesforceDroidGapActivity : CordovaActivity(), SalesforceActivityInterface {
```

Comment:
After the IDE's Kotlin conversion, I did a top-to-bottom review to polish the new Kotlin code.  I did stop short of any obvious logic changes and I also didn't re-order the class members.